esmvaltool/diag_scripts/bock20jgr/corr_pattern_collect.ncl

```diff
@@ -0,0 +1,475 @@
+    end do
+  end if
+
+  ; add lines for individual models
+  do ivar = 0, dimsizes(data_all(0, :))-1
+    do iproj = 0, dimsizes(projects)-1
+      ; Skip Project if no data for it
+      proj_mods = ind(data_all&models@project(mod_ind).eq.projects(iproj))
+      if .not. all(ismissing(data_all(proj_mods, ivar))) then
+        proj_center = x_val_proj(iproj, ivar)
+        xx = (/proj_center-l_length, proj_center+l_length/)
+        ; Plot lines for mean
+        xx_mean = (/proj_center-l_length*3., proj_center+l_length*3./)
+        yy_mean = (/means(iproj, ivar), means(iproj, ivar)/)
+        res_mlines@gsLineColor = fcolors(iproj, :)
+        res_lines@gsLineColor = fcolors(iproj, :)
+        plot@$unique_string("dum")$ = gsn_add_polyline( \
+          wks, plot, xx_mean, yy_mean, res_mlines)
+        do imod = 0, dimsizes(data_all(:, 0)) - 1
+          ; Only plot if model in right project
+          if data_all&models@project(imod).eq.projects(iproj) then
+            ; Don't plot obs as lines
+            if (.not.ismissing(data_all(imod, ivar))) then
+              if (data_all&models(imod).ne.alt_obs(ivar)) then
+                yy = (/data_all(imod, ivar), data_all(imod, ivar)/)
+                plot@$unique_string("dum")$ = gsn_add_polyline( \
+                  wks, plot, xx, yy, res_lines)
+                end if
+            end if
+          end if
+        end do
+        if (diag_script_info@plot_median) then
+          plot@$unique_string("dum_ci")$ = gsn_add_polymarker( \
+            wks, plot, x_val_proj(iproj, ivar), median(iproj, ivar), res_circw)
+        end if
+      end if
+      delete(proj_mods)
+    end do
+    if (alt_obs(ivar).ne."none") then
+      ; Plot obs as circles
+      plot@$unique_string("dum_circ")$ = gsn_add_polymarker( \
+        wks, plot, x_val(ivar), data_all(obs_ind(ivar), ivar), res_circ)
+    end if
+  end do
+
+  y_min_label = res@trYMinF + 0.1*yspan
+  lb_stride = yspan/res@vpHeightF * font_height * 2.
+  plabel = projects
+  ; Draw Legend
+  do iproj = 0, dimsizes(projects)-1
+    res_text@txFontColor = fcolors(iproj, :)
+    ; CMIP5 label has to be reduced to CMIP5 sometimes
+    if str_match_ind_ic(plabel, "CMIP5").eq.iproj then
+      plabel(iproj) = "CMIP5"
+    end if
+    plot@$unique_string("dum_l")$ = gsn_add_text(wks, plot, plabel(iproj),\
+                                                 x_val(0)-0.5, y_min_label + \
+                                                 lb_stride*(iproj+1.2), \
+                                                 res_text)
+  end do
+  res_text@txFontColor = res_circ@gsMarkerColor
+  plot@$unique_string("dum_l")$ = gsn_add_text(wks, plot, \
+                                               "alternative Observations", \
+                                               x_val(0)-0.5, y_min_label, \
+                                               res_text)
+
+  draw(plot)
+  frame(wks)
+  ; Write output
+  if config_user_info@write_netcdf then
```

Comment:
Note that the `write_netcdf` option will no longer be supported in the next version of the ESMValTool, meaning that this diagnostic script will crash.